CONTRIBUTING.md

```diff
@@ -0,0 +1,164 @@
+
+### 1. Component Standards
+
+##### A component must have the following files:
+
+- A folder within the `Components` directory named after the component family in camel case
+- The component file also in camel case
+- Stylesheet named after the component in camel case
+- Markdown documentation file named after the component in camel case
+- Storybook story named after the component in camel case
+
+##### Here's an example:
+
+```
+/Components
+  /MyComponent
+    MyComponent.tsx
+    MyComponent.scss
+    MyComponent.md
+    MyComponent.stories.tsx
+```
+
+- The component should import its associated stylesheet directly into itself
+  - The component stylesheet should import our modules file: `@import "../../Styles/modules.scss";`
+- The component (and all sub-components) should be imported & exported from `index.ts`
+  - Make sure to keep this list alphabetized for ease of use
+
+##### SCSS Conventions
+
+We use a variation on [BEM Naming Conventions](http://getbem.com/introduction/) which looks something like this:
+
+```css
+.component-name--element__modifier
+```
+```css
+// example:
+.button--checkbox__active
+```
+
+##### Function Components
+
+All Clockface components (with a few exceptions) are stateless and should be implemented as [Function Components](https://reactjs.org/docs/hooks-state.html#hooks-and-function-components)
+
+##### Standard Props
+
+We created a Typescript interface called `StandardProps` which ensures consistency across all components in the library. All components `Props` definitions should extend `StandardProps`.
+
+**All classnames must be prefixed with `cf-`**
+
+### 2. The `Base / Composed` Component Pattern
+
+In some cases it makes sense to have a fully customizable and unopinionated component. However we've noticed that these components tend to be used and customized the same way over and over. We adopted this pattern to offer ease of use without sacrificing the simplicity and extensibility of our components. You can see examples of this in action by looking at the `Button` family.
+
+- `Base` is the unopinionated and fully customizable component
+- `Composed` is a pre-customized (or "composed") version of a `Base` component, often with less options
+
+##### The folder structure for this pattern is:
+
+```
+/Components
+  /MyComponent
+    /Base
+      MyComponent.tsx
+      MyComponent.scss
+      MyComponent.md
+    /Composed
+      MyCoolComponent.tsx
+      MyCoolComponent.scss
+      MyCoolComponent.md
+      SimpleMyComponent.tsx
+      SimpleMyComponent.scss
+      SimpleMyComponent.md
+  MyComponent.stories.tsx
+```
+
+### 3. The Component Family Pattern
+
+We offer larger groups of components that are intended to be used together closely. In these cases we consider those components a "family". Component families not only share a namespace but are accessible via a single import. This is the only time we use classes.
+
+##### Here's an example:
+
+```tsx
+import {Apple} from './Apple'
```

Comment:
Imports here should be `FruitApple` and `FruitOrange`

Reply:
Ah good catch